src/bls/bls.service.ts

```diff
@@ -0,0 +1,105 @@
+import * as blst from '@chainsafe/blst';
+import { WINSTON_MODULE_NEST_PROVIDER } from 'nest-winston';
+import { Type, fromHexString } from '@chainsafe/ssz';
+import { ProviderService } from 'provider';
+import {
+  Inject,
+  Injectable,
+  OnModuleInit,
+  LoggerService,
+} from '@nestjs/common';
+import {
+  BLSPubkey,
+  Bytes32,
+  DOMAIN_DEPOSIT,
+  GENESIS_FORK_VERSION_BY_CHAIN_ID,
+  UintNum64,
+  ZERO_HASH,
+} from './bls.constants';
+import { DepositMessage, ForkData, SigningData } from './bls.containers';
+import { DepositData } from './interfaces';
+
+@Injectable()
+export class BlsService implements OnModuleInit {
+  constructor(
+    @Inject(WINSTON_MODULE_NEST_PROVIDER) private logger: LoggerService,
+    private providerService: ProviderService,
+  ) {}
+
+  async onModuleInit() {
+    const chainId = await this.providerService.getChainId();
+    this.forkVersion = GENESIS_FORK_VERSION_BY_CHAIN_ID[chainId] ?? null;
+  }
+
+  private forkVersion: Uint8Array | null = null;
+
+  public verify(depositData: DepositData): boolean {
```

Comment:
Untested

Reply:
<img width="815" alt="Screenshot 2022-07-14 at 15 18 13" src="https://user-images.githubusercontent.com/102730938/178980552-1d5a7f30-1e67-4d58-98f4-63c9cf586eec.png">

Reply:
Tests are in place, but only for basic functionality 

https://github.com/lidofinance/lido-council-daemon/blob/6fef341c2a83eba3f0beba89ec6692c178bce91d/src/bls/bls.service.spec.ts